distro/run/assembly/resources/run.bat

```diff
@@ -38,6 +38,19 @@ IF "x%JAVA_HOME%" == "x" (
   )
 )
 
+SET EXPECTED_JAVA_VERSION=17
+FOR /f "tokens=3" %%g IN ('JAVA -version 2^>^&1 ^| findstr /i "version"') DO (
```

Comment:
❓ Why do we need a for loop here and also in line 47? Can it loop more than once? Why does the last loop always yield the correct result?

Reply:
I will try to improve the script.

Reply:
The `FOR /F` [loop](https://ss64.com/nt/for_f.html) seems to be a standard way to find a string from a text. It loop is executed once from what I tested.
I didn't find a good alternative of the code. I am open for a suggestions, otherwise I will leave it as it is.